src/main/java/org/b612foundation/adam/opm/OdmScenarioBuilder.java

```diff
@@ -306,4 +310,168 @@ public static OrbitParameterMessage buildOpmWithCovariance(String hypercube) {
+    result.addBlock(block);
+    return result;
+  }
+
+  public static String getOemWithCovariance() {
+    return "CCSDS_OEM_VERS = 2.0\n"
+        + "CREATION_DATE = 1996-11-04T17:22:31\n"
+        + "ORIGINATOR = NASA/JPL\n"
+        + "\n"
+        + "META_START\n"
+        + "OBJECT_NAME          = MARS GLOBAL SURVEYOR\n"
+        + "OBJECT_ID            = 1996-062A\n"
+        + "CENTER_NAME          = SUN\n" // MARS BARYCENTER is not supported
+        + "REF_FRAME            = EME2000\n"
+        + "TIME_SYSTEM          = UTC\n"
+        + "START_TIME           = 1996-12-28T21:29:07.267\n"
+        + "USEABLE_START_TIME   = 1996-12-28T22:08:02.5\n"
+        + "USEABLE_STOP_TIME    = 1996-12-30T01:18:02.5\n"
+        + "STOP_TIME            = 1996-12-30T01:28:02.267\n"
+        + "INTERPOLATION        = HERMITE\n"
+        + "INTERPOLATION_DEGREE = 7\n"
+        + "META_STOP\n"
+        + "\n"
+        + "COMMENT  This block begins after trajectory correction maneuver TCM-3.\n"
+        + "1996-12-28T21:29:07.267 -2432.166 -063.042 1742.754  7.33702 -3.495867 -1.041945\n"
+        + "1996-12-28T21:59:02.267 -2445.234 -878.141 1873.073  1.86043 -3.421256 -0.996366\n"
+        + "1996-12-28T22:00:02.267 -2458.079 -683.858 2007.684  6.36786 -3.339563 -0.946654\n"
+        // intervening data records omitted here
+        + "1996-12-30T01:28:02.267 2164.375 1115.811 -688.131  -3.53328 -2.88452 0.88535\n"
+        + "\n"
+        + "COVARIANCE_START\n"
+        + "EPOCH = 1996-12-28T21:29:07.267\n"
+        + "COV_REF_FRAME = EME2000\n"
+        + "3.3313494e-04\n"
+        + "4.6189273e-04  6.7824216e-04\n"
+        + "-3.0700078e-04 -4.2212341e-04  3.2319319e-04\n"
+        + "-3.3493650e-07 -4.6860842e-07  2.4849495e-07  4.2960228e-10\n"
+        + "-2.2118325e-07 -2.8641868e-07  1.7980986e-07  2.6088992e-10  1.7675147e-10\n"
+        + "-3.0413460e-07 -4.9894969e-07  3.5403109e-07  1.8692631e-10  1.0088625e-10  6.2244443e-10\n"
+        + "\n"
+        + "EPOCH = 1996-12-29T21:00:00\n"
+        + "COV_REF_FRAME = EME2000\n"
+        + "3.4424505e-04\n"
+        + "4.5078162e-04  6.8935327e-04\n"
+        + "-3.0600067e-04 -4.1101230e-04  3.3420420e-04\n"
+        + "-3.2382549e-07 -4.5750731e-07  2.3738384e-07  4.3071339e-10\n"
+        + "-2.1007214e-07 -2.7530757e-07  1.6870875e-07  2.5077881e-10  1.8786258e-10\n"
+        + "-3.0302350e-07 -4.8783858e-07  3.4302008e-07  1.7581520e-10  1.0077514e-10  6.2244443e-10\n"
+        + "COVARIANCE_STOP";
+  }
+
+  public static OrbitEphemerisMessage buildOemWithCovariance() {
+    OrbitEphemerisMessage result = new OrbitEphemerisMessage();
+    result.setCcsds_oem_vers("2.0");
+    result.setHeader(new OdmCommonHeader().setCreation_date("1996-11-04T17:22:31").setOriginator("NASA/JPL"));
+
+    OemMetadata metadata = new OemMetadata();
+    metadata.setObject_name("MARS GLOBAL SURVEYOR");
+    metadata.setObject_id("1996-062A");
+    metadata.setCenter_name(CenterName.SUN); // MARS BARYCENTER
```

Comment:
I think you mean sun barycenter (in the comment)?

Reply:
Expanded the comment to explain. The example is copy-paste from the ODM standard, except there it's about Mars, and they use "MARS BARYCENTER" in this field in the example. We only support Sun and Earth in Adam (for now anyway), so I changed just this field without touching the rest. The resulting ephem doesn't make much sense, but this is a test for the parser only, not propagator.

Reply:
gotcha 👍